codecademy/practice_make_perfect.py

```diff
@@ -0,0 +1,105 @@
+import sys
+import math
+
+if sys.version_info > (3,):
+    long = int
+    unicode = str
+
+
+def is_int(x):
+    # function compute an integer number or not, and return false or true
+    assert isinstance(x, (int, float, long)) and type(x) is not bool, "{} passed the wrong data type".format(x)
+    if x - round(x) > 0:
+        return False
+    else:
+        if x < 0 and x - round(x) < 0:
+            return False
+        else:
+            return True
+# END is_int
+
+
+def factorial(x):
+    assert isinstance(x, (int, float, long)) and type(x) is not bool, "{} passed the wrong data type".format(x)
```

Comment:
cleaner would be:
`    assert type(x) in (int, float, long), "{} passed the wrong data type".format(x)`